### PR TITLE
Update and disable field validation on input object graph types progress

### PR DESCRIPTION
**Pull request by Shane32**
Please see issue #1383 

Validation code added and disabled (along with associated tests) until the issue with the SchemaBuilder has been fixed.  I'm not sure what needs to be done from this point.

### CODE REVIEW

File: src/GraphQL/Types/InputObjectGraphType.cs [hunk 1]
```diff
@@ -14,13 +14,23 @@ public class InputObjectGraphType<TSourceType> : ComplexGraphType<TSourceType>,
...
+
+                // throw new ArgumentException("InputObjectGraphType cannot have fields containing a ObjectGraphType.", nameof(fieldType.Type));
             }
 
             return base.AddField(fieldType);
         }
+
+        private static bool ValidateFieldType(FieldType fieldType)
+        {
+            if (fieldType.ResolvedType != null && fieldType.ResolvedType.IsInputType()) return true;
```
Comment (sungam3r) [inline src/GraphQL/Types/InputObjectGraphType.cs]:
```suggestion
            if (fieldType.ResolvedType?.IsInputType() == true) return true;
```

File: src/GraphQL/Types/InputObjectGraphType.cs [hunk 2]
```diff
@@ -14,13 +14,23 @@ public class InputObjectGraphType<TSourceType> : ComplexGraphType<TSourceType>,
...
+            if (fieldType.Type != null && fieldType.Type.IsInputType()) return true;
```
Comment (sungam3r) [inline src/GraphQL/Types/InputObjectGraphType.cs]:
```suggestion
            if (fieldType.Type?.IsInputType() == true) return true;
```

**Comment by Shane32** [comment]
Updated pursuant to pending comments in #1383

File: src/GraphQL/Types/InputObjectGraphType.cs [hunk 3]
```diff
@@ -14,13 +14,40 @@ public class InputObjectGraphType<TSourceType> : ComplexGraphType<TSourceType>,
     {
         public override FieldType AddField(FieldType fieldType)
         {
-            if (fieldType.Type == typeof(ObjectGraphType))
+            if (!ValidateFieldType(fieldType))
             {
                 throw new ArgumentException("InputObjectGraphType cannot have fields containing a ObjectGraphType.", nameof(fieldType.Type));
```
Comment (sungam3r) [inline src/GraphQL/Types/InputObjectGraphType.cs]:
```suggestion
                 throw new ArgumentException("InputObjectGraphType should contain only fields of 'input types' - enumerations, scalars and other InputObjectGraphTypes", nameof(fieldType.Type));
```

This error message is more informative.

**Comment by sungam3r** [comment]
Well, since we are adding such check, then let's make it complete - a similar check is necessary for the mirror case - when input field is added to the object type :).

**Comment by sungam3r** [comment]
Closed in favor of #1436

**Pull request closed without merging**